charts/theia-cloud/templates/instances-ingress-path-based.yaml

```
@@ -28,13 +28,24 @@ spec:
   tls:
   - hosts:
     - {{ tpl (.Values.hosts.configuration.baseHost | toString) . }}
-    {{- range .Values.hosts.allWildcardInstances }}
-    - {{ printf "'%s%s'" . $.Values.hosts.configuration.baseHost }}
-    {{- end }}
     {{- if .Values.ingress.addTLSSecretName }}
+    {{- range $wildcard := .Values.hosts.allWildcardInstances }}
+    {{- if (not (hasKey $.Values.ingress.allWildcardSecretNames $wildcard)) }}
+    - {{ tpl (printf "%s%s" . $.Values.hosts.configuration.baseHost | toString) . | quote }}

```

We should template the values separately. E.g. like this:
```suggestion
    - {{ printf "%s%s" (tpl . $) (tpl $.Values.hosts.configuration.baseHost $) | quote }}

```

---

Changed it 👍 As I'm fairly new to Helm templating, I don't see a reason for using `(tpl . $)` - could you briefly outline why it's necessary please? The `range` command returns the wildcard as string, right?

---

Just tested it out, `- {{ printf "%s%s" . (tpl $.Values.hosts.configuration.baseHost $)| quote }}` results in the same manifest in the context of `.` being `$wildcard`

---

It returns the same string as long as neither the wildcard nor $.Values.hosts.configuration.baseHost actually contain something to be templated. To be more precise, `tpl` runs the given string through the templating enging potentially replacing the value.

Assume you have these values:
```yml
config:
  test: "{{ .Values.templated }}"
  normal: "hello"
templated: "abc"
```

Now, evaluation the following in a helm template:
- `tpl .Values.config.test .` results in `abc`
- `tpl .Values.config.normal .` results in `hello`

You can see that the `tpl` function evaluates the `{{ .Values.templated }}`. but for normal values the value is taken as is.
Thus, you don't see a difference in the approaches when using plain values. However, thinking about this even more, `tpl` can probably also resolve a string hat contains two templates. In any case replacing the `.` with `$` as the second argument for `tpl` was necessary because inside a range `.` is not the root context anymore but the current iteration value.